pipeline/cloud/http.py

```diff
@@ -17,29 +16,55 @@
 _client_async = None
 
 
-def get_response_error_dict(e: httpx.HTTPStatusError) -> t.Dict:
+class APIError(Exception):
+    def __init__(
+        self, url: str, status_code: int, detail: dict | str, request_id: str | None
+    ):
+        self.url = url
+        self.status_code = status_code
+        self.detail = detail
+        self.request_id = request_id
+        super().__init__(url, status_code, detail, request_id)
+
+    @classmethod
+    def from_response(cls, response: httpx.Response):
+        try:
+            detail = response.json()["detail"]
+            if not isinstance(detail, dict):
+                detail = {"detail": detail}
+        except (JSONDecodeError, TypeError, KeyError):
+            detail = {
+                "message": "Something went wrong.",
+                "response": response.text,
+            }
+        return cls(
+            url=str(response.url),
+            detail=detail,
+            status_code=response.status_code,
+            request_id=response.headers.get("x-correlation-id"),
+        )
+
+    def __str__(self):
+        error_msg = ", ".join(f"{k}={v}" for k, v in self.__dict__.items())
+        error_msg = f"APIError({error_msg})"
+        return error_msg
+
+
+def raise_if_http_status_error(response: httpx.Response):
     try:
-        detail = e.response.json()["detail"]
-        if not isinstance(detail, dict):
-            detail = {"detail": detail}
-    except (TypeError, KeyError):
-        return {"message": "Something went wrong.", "response_json": e.response.json()}
-    return {**detail, "request_id": e.response.headers.get("x-correlation-id")}
+        response.raise_for_status()
+    except httpx.HTTPStatusError as e:
+        error = APIError.from_response(e.response)
+        _print(str(error), level="ERROR")
+        raise error from None
```

Comment:
Ah nice so will this suppress the traceback from the `HTTPStatusError`?

Reply:
yeah, exactly. Only just learnt this :)